### DISCUSSION TITLE
Adafruit RFM69HCW Transceiver Radio Bonnet

### DISCUSSION THREAD

**Discussion by chrilis** [QuadCorei8085/elero_protocol#3] [General]
Hi @QuadCorei8085,

thank you for your work on this repository! The following is a questions and not an issue, so you may want to enable discussions and move it there.

I have two VarioTel 2 controlling two and four blinds respectively. Using the [Adafruit RFM69HCW Transceiver Radio Bonnet](https://www.adafruit.com/product/4073) with a Rasperberry Pi 4 and a mix of your code and basic RFM69 Python tutorials I managed to receive some signals.
My current problem is that I receive a lot of different signals. I assume that is because the system works bidrectional and the blinds communicate among each other. Additionally the decoded payloads are inconsistent for the same buttons (e.g. pressing _up_ multiple times results in very different payloads). So I wonder if I have done something fundamentally wrong.

If you have the time, I would be very happy if I could send you some received signals and parts of the code so you can look over it.

Many thanks again! 🙂

**Comment by QuadCorei8085** [comment]
Sure I would be interested to see other systems. I'm not familiar with all features of github, is there a way to send a private message? I think its better to just send me a text file in email.

**Comment by QuadCorei8085** [comment]
Or just post the records on pastebin or similar site and send the URL here.

**Comment by QuadCorei8085** [comment]
Also if I remember correctly from long time ago these RFM69 modules are actually a Silicon Labs radio Si4432(?) so make sure your radio settings match the CC1100 settings (Texas Instruments). I'm not familiar with the details as I only grabbed the config and replayed it to my radio over SPI but since these are digital radios its not only the frequency bands, modulations, but packet format and CRC needs to match for "correct" reception. Certainly possible to tune one to another - I personally didn't want to spend time debugging and ordered the similar radio IC.

**Comment by chrilis** [comment]
> I'm not familiar with all features of github, is there a way to send a private message? I think its better to just send me a text file in email.

Here are instructions to [enable discussions](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/enabling-or-disabling-github-discussions-for-a-repository#enabling-or-disabling-github-discussions-for-your-repository) and [convert an issue to a discussion](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#converting-an-issue-to-a-discussion).

> Or just post the records on pastebin or similar site and send the URL here.

I recorded signals for the buttons _Up_, _Down_ and _Stop_. For each remote I pressed the buttons three times. I believe four signals are sent for _Up_ and _Down_ (two identical signals for pressing the button and two more identical signals for releasing it). For _Stop_, I think only one signal is sent. Please let me know if you need other or more information per signal.

<details>
<summary>Up</summary>

```
[2022-10-30 09:47:18] packet received from remote-bedroom
29 pck_len=0x1B 0x52 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xE9 0x79 0x6B 0x92 0x8F 0x0D 0x96 0x25 

[2022-10-30 09:47:18] packet received from remote-bedroom
29 pck_len=0x1B 0x52 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xE9 0x79 0x6B 0x92 0x8F 0x0D 0x96 0x25 

[2022-10-30 09:47:18] packet received from remote-bedroom
29 pck_len=0x1B 0x53 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x39 0x36 0xAB 0x7E 0xAF 0xB4 0xC6 0xEA 

[2022-10-30 09:47:18] packet received from remote-bedroom
29 pck_len=0x1B 0x53 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x39 0x36 0xAB 0x7E 0xAF 0xB4 0xC6 0xEA 

[2022-10-30 09:47:25] packet received from remote-bedroom
29 pck_len=0x1B 0x50 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x63 0xEB 0x3E 0x64 0x1E 0xA6 0x1C 0xB6 

[2022-10-30 09:47:25] packet received from remote-bedroom
29 pck_len=0x1B 0x50 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x63 0xEB 0x3E 0x64 0x1E 0xA6 0x1C 0xB6 

[2022-10-30 09:47:25] packet received from remote-bedroom
29 pck_len=0x1B 0x51 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x83 0x05 0xFE 0xED 0xEE 0x3B 0x8C 0x68 

[2022-10-30 09:47:25] packet received from remote-bedroom
29 pck_len=0x1B 0x51 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x83 0x05 0xFE 0xED 0xEE 0x3B 0x8C 0x68 

[2022-10-30 09:47:32] packet received from remote-bedroom
29 pck_len=0x1B 0x5E 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x5D 0x5F 0xDD 0x5A 0xF4 0x81 0xF5 0x09 

[2022-10-30 09:47:32] packet received from remote-bedroom
29 pck_len=0x1B 0x5E 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x5D 0x5F 0xDD 0x5A 0xF4 0x81 0xF5 0x09 

[2022-10-30 09:47:32] packet received from remote-bedroom
29 pck_len=0x1B 0x5F 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x10 0x80 0xC7 0x05 0xC6 0xCE 0xE9 0x50 

[2022-10-30 09:47:32] packet received from remote-bedroom
29 pck_len=0x1B 0x5F 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x10 0x80 0xC7 0x05 0xC6 0xCE 0xE9 0x50 

[2022-10-30 09:47:40] packet received from remote-living-room
29 pck_len=0x1B 0xAF 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x50 0x23 0xD7 0xAC 0xF6 0x77 0xF9 0x5B 

[2022-10-30 09:47:40] packet received from remote-living-room
29 pck_len=0x1B 0xAF 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x50 0x23 0xD7 0xAC 0xF6 0x77 0xF9 0x5B 

[2022-10-30 09:47:40] packet received from remote-living-room
29 pck_len=0x1B 0xAC 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x10 0xDA 0xC7 0xD3 0xC6 0x98 0xE9 0x47 

[2022-10-30 09:47:40] packet received from remote-living-room
29 pck_len=0x1B 0xAC 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x10 0xDA 0xC7 0xD3 0xC6 0x98 0xE9 0x47 

[2022-10-30 09:47:45] packet received from remote-living-room
29 pck_len=0x1B 0xAD 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xFF 0xC8 0x89 0xFF 0x6D 0xF5 0x00 0xDD 

[2022-10-30 09:47:45] packet received from remote-living-room
29 pck_len=0x1B 0xAD 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xFF 0xC8 0x89 0xFF 0x6D 0xF5 0x00 0xDD 

[2022-10-30 09:47:45] packet received from remote-living-room
29 pck_len=0x1B 0xAA 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xDF 0xF4 0x69 0xB1 0xBD 0x2A 0xB0 0x24 

[2022-10-30 09:47:45] packet received from remote-living-room
29 pck_len=0x1B 0xAA 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xDF 0xF4 0x69 0xB1 0xBD 0x2A 0xB0 0x24 

[2022-10-30 09:47:51] packet received from remote-living-room
29 pck_len=0x1B 0xAB 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xC8 0x41 0x28 0xC0 0x98 0x1F 0x77 0x33 

[2022-10-30 09:47:51] packet received from remote-living-room
29 pck_len=0x1B 0xAB 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xC8 0x41 0x28 0xC0 0x98 0x1F 0x77 0x33 

[2022-10-30 09:47:51] packet received from remote-living-room
29 pck_len=0x1B 0xA8 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xE1 0x1D 0x92 0x29 0x8B 0x63 0x92 0x8C 

[2022-10-30 09:47:51] packet received from remote-living-room
29 pck_len=0x1B 0xA8 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0xE1 0x1D 0x92 0x29 0x8B 0x63 0x92 0x8C
```

</details>

<details>
<summary>Down</summary>

```
[2022-10-30 09:48:55] packet received from remote-bedroom
29 pck_len=0x1B 0x5C 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x00 0xAC 0xD7 0x18 0x76 0xD2 0xA9 0x42 

[2022-10-30 09:48:55] packet received from remote-bedroom
29 pck_len=0x1B 0x5C 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x00 0xAC 0xD7 0x18 0x76 0xD2 0xA9 0x42 

[2022-10-30 09:48:56] packet received from remote-bedroom
29 pck_len=0x1B 0x5D 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xDF 0xB2 0x69 0x86 0xBD 0x50 0xB0 0x1E 

[2022-10-30 09:48:56] packet received from remote-bedroom
29 pck_len=0x1B 0x5D 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xDF 0xB2 0x69 0x86 0xBD 0x50 0xB0 0x1E 

[2022-10-30 09:49:02] packet received from remote-bedroom
29 pck_len=0x1B 0x5A 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xCF 0x93 0x49 0x3C 0x9D 0xE7 0x70 0xCB 

[2022-10-30 09:49:02] packet received from remote-bedroom
29 pck_len=0x1B 0x5A 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xCF 0x93 0x49 0x3C 0x9D 0xE7 0x70 0xCB 

[2022-10-30 09:49:02] packet received from remote-bedroom
29 pck_len=0x1B 0x5B 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xE8 0x2A 0x98 0xA3 0x88 0x78 0x97 0x37 

[2022-10-30 09:49:02] packet received from remote-bedroom
29 pck_len=0x1B 0x5B 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0xE8 0x2A 0x98 0xA3 0x88 0x78 0x97 0x37 

[2022-10-30 09:49:08] packet received from remote-bedroom
29 pck_len=0x1B 0x58 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x38 0xD8 0x28 0xDF 0xA8 0x95 0xC7 0x8D 

[2022-10-30 09:49:08] packet received from remote-bedroom
29 pck_len=0x1B 0x58 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x38 0xD8 0x28 0xDF 0xA8 0x95 0xC7 0x8D 

[2022-10-30 09:49:08] packet received from remote-bedroom
29 pck_len=0x1B 0x59 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x61 0xC4 0x12 0xF1 0x1B 0xFA 0x12 0x64 

[2022-10-30 09:49:08] packet received from remote-bedroom
29 pck_len=0x1B 0x59 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x61 0xC4 0x12 0xF1 0x1B 0xFA 0x12 0x64 

[2022-10-30 09:49:16] packet received from remote-living-room
29 pck_len=0x1B 0xA9 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x31 0x67 0x22 0x4B 0xAB 0x49 0xC2 0x71 

[2022-10-30 09:49:16] packet received from remote-living-room
29 pck_len=0x1B 0xA9 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x31 0x67 0x22 0x4B 0xAB 0x49 0xC2 0x71 

[2022-10-30 09:49:16] packet received from remote-living-room
29 pck_len=0x1B 0xB6 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x6B 0xBE 0x10 0x87 0x11 0x5C 0x1F 0x1F 

[2022-10-30 09:49:16] packet received from remote-living-room
29 pck_len=0x1B 0xB6 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x6B 0xBE 0x10 0x87 0x11 0x5C 0x1F 0x1F 

[2022-10-30 09:49:21] packet received from remote-living-room
29 pck_len=0x1B 0xB7 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x8B 0x99 0xB0 0x32 0xE1 0xED 0x8F 0x25 

[2022-10-30 09:49:21] packet received from remote-living-room
29 pck_len=0x1B 0xB7 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x8B 0x99 0xB0 0x32 0xE1 0xED 0x8F 0x25 

[2022-10-30 09:49:21] packet received from remote-living-room
29 pck_len=0x1B 0xB4 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x57 0x26 0xE5 0xAE 0xF5 0x74 0xF8 0xEA 

[2022-10-30 09:49:21] packet received from remote-living-room
29 pck_len=0x1B 0xB4 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x57 0x26 0xE5 0xAE 0xF5 0x74 0xF8 0xEA 

[2022-10-30 09:49:27] packet received from remote-living-room
29 pck_len=0x1B 0xB5 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x17 0xDB 0x05 0xD4 0xC5 0x96 0xE8 0xB6 

[2022-10-30 09:49:27] packet received from remote-living-room
29 pck_len=0x1B 0xB5 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x17 0xDB 0x05 0xD4 0xC5 0x96 0xE8 0xB6 

[2022-10-30 09:49:28] packet received from remote-living-room
29 pck_len=0x1B 0xB2 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x02 0xC5 0x7F 0xFD 0x72 0xFB 0xAD 0x68 

[2022-10-30 09:49:28] packet received from remote-living-room
29 pck_len=0x1B 0xB2 0xE1 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xC1 0x99 0xC0 0x02 0xC5 0x7F 0xFD 0x72 0xFB 0xAD 0x68 
```

</details>

<details>
<summary>Stop</summary>

```
[2022-10-30 09:50:31] packet received from ?
29 pck_len=0x1B 0x9C 0xE0 0xD3 0x0F 0x25 0xFE 0xCA 0x2E 0x17 0xA5 0x6C 0x4B 0xB4 0x2E 0x69 0x7F 0xC2 0x99 0x81 0x51 0x6A 0xD2 0x49 0xFB 0x48 0xFF 0xB7 

[2022-10-30 09:50:39] packet received from remote-bedroom
31 pck_len=0x1D 0x26 0xCF 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xF4 0x50 0xF6 0xDB 0xBE 0xBD 0x32 0x74 0x0D 0x4E 0x0B 0xE4 0x5C 

[2022-10-30 09:50:39] packet received from ?
31 pck_len=0x1D 0x54 0x6F 0xD3 0x50 0x25 0xBF 0x37 0x74 0x11 0xAE 0x6C 0x4B 0x85 0x6C 0x5B 0x7F 0x38 0x48 0xC2 0xDA 0x1C 0x8A 0xC5 0x2A 0x53 0x79 0x10 0x5A 0xB5 

[2022-10-30 09:50:45] packet received from remote-bedroom
31 pck_len=0x1D 0x27 0xCF 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xF4 0x50 0xF6 0xDB 0xBE 0x67 0x8E 0xC6 0x74 0x54 0x37 0x99 0x13 

[2022-10-30 09:50:51] packet received from remote-bedroom
31 pck_len=0x1D 0x24 0xCF 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xF4 0x50 0xF6 0xDB 0xBE 0x2B 0xD4 0xF3 0x96 0x60 0x4D 0x8E 0xBE 

[2022-10-30 09:50:59] packet received from remote-living-room
31 pck_len=0x1D 0xB0 0xCF 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xFF 0x50 0xF6 0xDB 0xBE 0x12 0x83 0x40 0x78 0xF2 0x3A 0x38 0x1F 

[2022-10-30 09:50:59] packet received from ?
31 pck_len=0x1D 0x5E 0x6F 0xD3 0x50 0x25 0x29 0x3C 0x74 0x11 0xA5 0x6C 0x4B 0x41 0x5D 0x5B 0x7F 0xFC 0x79 0xC2 0xDA 0x1C 0xD1 0xE1 0xFB 0x1B 0x21 0x54 0xF9 0xD1 

[2022-10-30 09:51:05] packet received from remote-living-room
31 pck_len=0x1D 0xB1 0xCF 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xFF 0x50 0xF6 0xDB 0xBE 0xA0 0xDF 0x5C 0x95 0xEF 0x46 0x02 0xBD 

[2022-10-30 09:51:05] packet received from ?
31 pck_len=0x1D 0x5F 0x6F 0xD3 0x50 0x25 0x28 0x3C 0x74 0x11 0xA5 0x6C 0x4B 0x41 0x5D 0x5B 0x7F 0xFC 0x79 0xC2 0xDA 0x1C 0x01 0x10 0xCB 0x61 0x01 0xB3 0xA9 0x94 

[2022-10-30 09:51:06] packet received from ?
29 pck_len=0x1B 0xB0 0xE0 0xD3 0x0F 0x25 0xFE 0xAF 0xA6 0x3B 0xA5 0x6C 0x4B 0xD1 0xA6 0x45 0x7F 0xC2 0x99 0x81 0x24 0x40 0x74 0xC9 0x59 0x1E 0x5B 0x50 

[2022-10-30 09:51:10] packet received from remote-living-room
31 pck_len=0x1D 0xBE 0xCF 0xD3 0x5A 0x25 0xFD 0x3F 0x5D 0x25 0xA6 0x45 0x7F 0x41 0x5D 0x5B 0x7F 0xFF 0x50 0xF6 0xDB 0xBE 0x90 0xA1 0x2C 0xFB 0xAF 0x64 0xB2 0x81 

[2022-10-30 09:51:10] packet received from ?
31 pck_len=0x1D 0x5C 0x6F 0xD3 0x50 0x25 0x27 0x3C 0x74 0x11 0xA5 0x6C 0x4B 0x41 0x5D 0x5B 0x7F 0xFC 0x79 0xC2 0xDA 0x1C 0x5C 0x09 0x71 0x4E 0xB3 0xDC 0x74 0xB8 
```

</details>

I also tried debugging the signals and outputting them with more (and interpreted) information. I paste an example below, but I assume most of the interpretation is not correct 🙈.

<details>
<summary>Extended signal information - Up</summary>
  
```
[2022-10-30 10:28:49] packet received from remote-bedroom
key                hex                                                  decimal  comment
-----------------  -------------------------------------------------  ---------  -------------------------
length             0x1B                                                      27
packet_count       0x25                                                      37
packet_info_1      0xE1                                                     225
packet_info_2      0xD3                                                     211
hop_info           0x5A                                                      90
system_address     0x25                                                      37
source_group       0xEE                                                     238  channel 01
source_address     0xFB 0x6C 0x25                                                remote-bedroom
backward_address   0x62 0x74 0x7F                                                <- motor-bedroom
forward_address    0x85 0x6C 0x5B                                                -> motor-work-room-garden
destination_count  0x7F                                                     127
destination        0xD2                                                     210  group 02
payload            0x99 0xC0 0x07 0xAE 0x55 0x17 0x75 0xDC 0xA8 0x4F
payload_decoded    0x00 0x00 0x15 0x47 0x69 0x7C 0xD6 0x2D                       0b00010101=up
29 pck_len=0x1B 0x25 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0x62 0x74 0x7F 0x85 0x6C 0x5B 0x7F 0xD2 0x99 0xC0 0x07 0xAE 0x55 0x17 0x75 0xDC 0xA8 0x4F 
29 pck_len=0x1B 0x26 0xE1 0xD3 0x5A 0x25 0xEE 0xFB 0x6C 0x25 0xFB 0x6C 0x25 0xFB 0x6C 0x25 0x01 0xD2 0x99 0xC0 0x6B 0x43 0x9E 0x2C 0xCE 0x87 0xFD 0xF7
```
  
</details>

> Also if I remember correctly from long time ago these RFM69 modules are actually a Silicon Labs radio Si4432(?) so make sure your radio settings match the CC1100 settings (Texas Instruments)

I used this [datasheet](https://cdn.sparkfun.com/datasheets/Wireless/General/RFM69HCW-V1.1.pdf) and _tried_ to use the same config as in your [elero_cc1100_init](https://github.com/QuadCorei8085/elero_protocol/blob/91d63c6bf3735f7a18305f31562005fef209914b/main.cpp#L712) method. Maybe you can tell from the signals above, if the settings could be correct.

Please let me know if I can provide you with any other useful information. Thank you very much!

**Comment by QuadCorei8085** [comment]
Just had a look at it, first thing I noticed in your messages (compared to my recordings). 

1) Whenever a remote (variotel2) sends a message in my case it is 3x repeated (for UP and DOWN commands). 

2) The SRC DST and HOP/SENDER address is always the same (in my case). In your messages I see no repeating address fields (all addresses are different, which is strange - at least the SRC and DST should be the same when a blind repeats the mesage it just replaces the SENDER/HOP address with its own).

3) In your UP records the CNT field is incremented strangely. The first pair is with 0x52 and 0x53 respectively but then it goes back to 0x50 and 0x51 and then 0x5E 0x5F?, then there is a packet with index 0xAF followed by 0xAC. Really strange. Either this variotel works somehow fundamentally different or the radio settings are not equivalent?

Are you sure the bitstream of the radio is captured correctly? Strange that the packet length seems correct on both cases 0x1B for up/down and 0x1D for stop, but the rest doesnt match my setup (i have 3 remotes, all working the same way).

**Reply by chrilis** [reply]
Thank you for your quick reply! The fact that the CNT is incorrect and I'm apperently missing some messages, leads me to believe that the radio settings are not correct. Although my knowledge about radios is (unfortunately) extremely limited, I will try to list a few of the current settings:
- frequency: `869.525`
- frequency deviation: `34912`
- preamble length: `12`
- baudrate: `76767`
- bitrate: `76767`
- sync word: `0xd3 0x91 0xd3 0x91`
- modulation type: `FSK`
- modulation shape: `Gaussian filter, BT=0.5`
- packet format: `Variable length`
- dc-free encoding: `Whitening`

Maybe there is already something wrong with the above values. If not, I will have to compare the CC1101 and the RFM69HCW datasheets and the values in your repo once again 😄

**Comment by QuadCorei8085** [comment]
I unfortunately never checked/decoded the radio settings. I once tried to use the CC1101 for another project and I vaguely remember there was an editor/configurator that spit out the register settings for a given setup. My impression from those bits and values that it is highly likely to missinterpret the actual configuration based on trying to reverse engineer the settings by crosschecking it with the datasheet, maybe you could use that TI code/register generator and input the register values and observe the configuration (if it works updates the config when register is changed).
I think it was the SmartRF studio: https://www.ti.com/product/CC1101#design-tools-simulation you could give it a try.  I personally wouldn't like to go this way.

For me the easiest would be to use a CC1100 or CC1101 module from ebay/aliexpress/amazon. The cost of such modules is much cheaper than grinding nerves :)

**Reply by chrilis** [reply]
Thanks for your advice! I tried to use the SmartRF studio you recommended and I entered the register settings of your repo, but the output of the tool was less helpful than I hoped. But the radio configuration must be wrong, because I can't imagine my VarioTel behaving so differently from yours. 

So maybe I will order an ESP32 and a CC1101. Thanks again! 🙂

**Reply by QuadCorei8085** [reply]
Probably that is the easiest. 

I see this bonnet is for RPi, there is an official USB dongle from Elero that (i believe using serial commands) can control 12 channels or so. More expensive of course and only 12 channel. 
For an RPi you can just buy the CC1101 module, I guess there are also python/C codes for those.

Just one note on the HW side, the modules are 868 and i'm pretty sure those you can buy cheap have some kind of defect on the antenna matching side and the antenna that it comes with also not the best. Combined with a wrongly placed/shaded reciever/blind antenna it can be that (if you manage to put it together) you can control a few of them and some won't respond.
It took me a while to figure out that my transmitter antenna wasnt tuned for the frequency I wanted to use, fortunately had access to VNA.
But simply if you ever get there and you find some are not responding just move the CC1101 module closer and try it there.
If you see it works in close range you should check the antennas :)

**Reply by chrilis** [reply]
Do you think this [module](https://www.ebay.de/itm/255283295154) would work @QuadCorei8085? It says 433MHz but shouldn't the CC1101 support 868 MHz too?
I haven't found a module with an SPI adapter that says 868MHz and is in stock yet. Unfortunately buying a module without adapter is not an option for me as I can't  solder at all 🙈

Thanks again for your help!

**Reply by QuadCorei8085** [reply]
The CC1101 is a programmable radio so the SW will program/tune it to 868. 

However those tiny components between the CC1101 and the antenna are the matching/filtering "network" they are there to filter out anything outside the tuned (in this case 433mhz) frequency. 
In the datasheet (my copy) shows it in Figure 10 and 11 on page 25/98 the difference.

A filter in reality however not a perfect "no pass", it will allow signals in and out in the unwanted frequencies, but extremely attenuated. Hard to tell without measuring the filter, it might work from 1cm or from 10cm.

Might be only good to check out your code integration but without swapping out those tiny components you will not achieve a decent range.

**Reply by chrilis** [reply]
Thank you for your detailed explanation, once again! So I guess I have to find an available adapter or finally try to learn how to solder 😅

**Reply by QuadCorei8085** [reply]
Checking now my modules I found that for testing I actually used a 433 module. And it worked well in my apartment (10-15m) but I used an antenna tuned for 868 so if you have an antenna you could use the 433 with this limited range in mind.

**Reply by chrilis** [reply]
Good to know! I do have an 863-928MHz antenna, so I might give the 443MHz module a try. 10 to 15m should be fine, at least to get started.
Thank you 😊

**Reply by chrilis** [reply]
@QuadCorei8085 I just wanted to let you know that the 443MHz module works pretty well. Currently I can control 4 out of 6 blinds with this [repo](https://github.com/stanleypa/eleropy). Thanks again for your help!
